components/nimbus/src/json.rs

```diff
@@ -0,0 +1,183 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use serde_json::{Map, Value};
+use std::collections::HashMap;
+
+/// Replace any instance of [from] with [to] in any string within the [serde_json::Value].
+///
+/// This recursively descends into the object, looking at string values and keys.
+#[allow(dead_code)]
+pub(crate) fn replace_str(value: &mut Value, from: &str, to: &str) {
+    let replacer = |s: &str| -> Option<String> {
+        if s.contains(from) {
+            Some(s.replace(from, to))
+        } else {
+            None
+        }
+    };
+    replace_str_with(value, &replacer);
+}
+
+/// Replace any instance of [from] with [to] in any string within the [serde_json::Value::Map].
+///
+/// This recursively descends into the object, looking at string values and keys.
+pub(crate) fn replace_str_in_map(map: &mut Map<String, Value>, from: &str, to: &str) {
+    let replacer = |s: &str| -> Option<String> {
+        if s.contains(from) {
+            Some(s.replace(from, to))
+        } else {
+            None
+        }
+    };
```

Comment:
It looks like this code is written the same in a couple places — could it be refactored out into the file scope?